R/semiparametricsurvivalanalysis.R

```diff
@@ -228,34 +454,363 @@ SemiParametricSurvivalAnalysis <- function(jaspResults, dataset, options, state
+
+  if (jaspBase::isTryError(fitTest)) {
+    proportionalHazardsTable$setError(gettextf("The model test failed with the following message: %1$s", fitTest))
+    return()
+  }
+
+  fitTestSummary <- cbind.data.frame(param = rownames(fitTest[["table"]]), fitTest[["table"]])
+  fitTestSummary$param[fitTestSummary$param == "GLOBAL"] <- gettext("Global")
+  fitTestSummary$param <- sapply(fitTestSummary$param, function(x) .saTermNames(x, c(options[["covariates"]], options[["factors"]])))
+
+  proportionalHazardsTable$setData(fitTestSummary)
+  proportionalHazardsTable$addFootnote(gettextf("Tests are performed using the %1$s survival times transformation.", options[["proportionalHazardsTransformation"]]))
+
+  return()
+}
+.saspProportionalHazardsPlots <- function(jaspResults, dataset, options) {
+
+  if (!is.null(jaspResults[["proportionalHazardsPlots"]]))
+    return()
+
+  proportionalHazardsPlots <- createJaspContainer(title = gettext("Proporational Hazards Assumption Test Plots"))
+  proportionalHazardsPlots$dependOn(c(.saspDependencies, "proportionalHazardsPlot", "proportionalHazardsTransformation", "proportionalHazardsTestTerms"))
+  proportionalHazardsPlots$position <- 8
+  jaspResults[["proportionalHazardsPlots"]] <- proportionalHazardsPlots
+
+  if (!.saSurvivalReady(options)) {
+    surivalPlot <- createJaspPlot()
+    proportionalHazardsPlots[["waitingPlot"]] <- surivalPlot
+    return()
+  }
+
+  fitTest <- jaspResults[["fitTest"]][["object"]]
+
+  if (jaspBase::isTryError(fitTest)) {
+    surivalPlot <- createJaspPlot()
+    surivalPlot$setError(gettextf("The model test failed with the following message: %1$s", fitTest))
+    proportionalHazardsPlots[["waitingPlot"]] <- surivalPlot
+    return()
+  }
+
+  for (i in 1:(nrow(fitTest$table) - 1)) {
+
+    tempVariable    <- rownames(fitTest$table)[i]
+    tempFitTestPlot <- plot(fitTest, plot = FALSE, var = tempVariable)
+
+    # adapted from the survival:::plot.cox.zph
+    tempDfPoints <- data.frame(
+      x = fitTest$x,
+      y = fitTest$y[,i]
+    )
+    tempDfPrediction <- data.frame(
+      x = tempFitTestPlot$x,
+      y = tempFitTestPlot$y[,1]
+    )
+    tempDfCiBand <- data.frame(
+        x = c(tempFitTestPlot$x,     rev(tempFitTestPlot$x)),
+        y = c(tempFitTestPlot$y[,2], rev(tempFitTestPlot$y[,3]))
+    )
+
+    if (fitTest$transform == "log") {
+      tempDfPoints$x <- exp(tempDfPoints$x)
+    }
+
+    # x-ticks
+    xTime <- fitTest$time
+    indx  <- !duplicated(tempDfPoints$x)
+    aprX  <- approx(tempDfPoints$x[indx], xTime[indx], seq(min(tempDfPoints$x), max(tempDfPoints$x), length = 5))
+
+    # y-ticks
+    yTicks <- jaspGraphs::getPrettyAxisBreaks(range(c(tempDfPoints$y, tempDfCiBand$y)))
+
+    tempPlot <- ggplot2::ggplot() +
+      ggplot2::geom_polygon(data = tempDfCiBand, mapping = ggplot2::aes(x = x, y = y), fill = "grey", alpha = 0.5) +
+      jaspGraphs::geom_line(data = tempDfPrediction, mapping = ggplot2::aes(x = x, y = y)) +
```

Comment:
just a question, why use jaspGRaphs? What differs to ggplot2?

Reply:
The default linewidth is different